pkgs/development/python-modules/pypdf2/default.nix

```diff
@@ -22,9 +21,10 @@ buildPythonPackage rec {
 
   nativeBuildInputs = [ flit-core ];
 
-  propagatedBuildInputs = lib.optionals (pythonOlder "3.10") [ typing-extensions ];
+  dependencies = lib.optionals (pythonOlder "3.10") [ typing-extensions ];
 
-  nativeCheckInputs = [ unittestCheckHook ];
+  # no test
```

Comment:
There are tests https://github.com/py-pdf/pypdf/tree/main/tests

We probably need to fetch from GitHub to get them

Reply:
`PyPDF2` back to  `PyPDF` after `v3.0.1`, but there’s no tag for that version in the `pypdf` repo.

I compared the source package downloaded from [PyPi](https://files.pythonhosted.org/packages/9f/bb/18dc3062d37db6c491392007dfd1a7f524bb95886eb956569ac38a23a784/PyPDF2-3.0.1.tar.gz) with the contents of pypdf [v3.0.0](https://github.com/py-pdf/pypdf/tree/3214dde1d2c0f7c9071ba4157d82cdb796dd73d4), and the only difference is that the source package from PyPi contains a warning about pypdf2 being deprecated. 

```
+warnings.warn(
+    message="PyPDF2 is deprecated. Please move to the pypdf library instead.",
+    category=DeprecationWarning,
+)
```

Should we downgrade it to version 3.0.0 and switch the source to GitHub?

Reply:
@SuperSandro2000 
I believe we didn’t need these tests anymore. Running `pytest` (if configured to ignore the deprecation warnings) on patched 3.0.0 resulted in 184 errors, over 160 related to network issue, with the rest being `FileNotFound`.

Considering that `PyPDF2` has already been marked as deprecated upstream and is not actively maintained, we don't need to spend much effort on this.